docs/implementation.adoc

```diff
@@ -0,0 +1,507 @@
+== Implementation Guide
+
+This document provides the guidelines for a correct optimal implementation of the SCM API.
+
+=== Introduction
+
+The SCM API consists of a number of extension points:
+
+* The original `hudson.scm.SCM` extension point is responsible for:
+
+** performing checkouts from source control into a `FilePath`
+
+** calculating the changelogs between revisions
+
+** constructing the changelog parser
```

Comment:
I think you can construct whatever change log you want by using repeated calls to `SCMFileSystem.of(SCMSource, SCMHead, SCMRevision)` followed by `SCMFileSystem.changesSince(SCMRevision, OutputStream)`.

To do anything more advanced would requiring enforcing a contract on the changelog entries which - while something that would be really nice to have - is not something I have the energy to try and drive through